### PR TITLE
[jaeger-v2] Implement empty service name sanitizer for OTLP

### PR DESCRIPTION
**Pull request by mahadzaryab1**
## Which problem is this PR solving?
- Towards #5545

## Description of the changes
- Created a new `Sanitizer` type for sanitizing traces in OTLP format 
- Implemented the empty service name sanitizer; I'll open a separate PR for the UTF-8 sanitizer

## How was this change tested?
- Unit tests

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by yurishkuro** [comment]
do not capitalize all words in PR titles, only the first one (i.e. normal sentence case).

File: cmd/jaeger/internal/exporters/storageexporter/exporter.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 	"go.opentelemetry.io/collector/pdata/ptrace"
 	"go.uber.org/zap"
 
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/exporters/storageexporter/sanitizer"
```
Comment (yurishkuro) [inline cmd/jaeger/internal/exporters/storageexporter/exporter.go:14]: simpler?
```suggestion
	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/sanitizer"
```

File: cmd/jaeger/internal/exporters/storageexporter/sanitizer/sanitizer.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package sanitizer
+
+import (
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// Sanitize is a function that performs enrichment, clean-up, or normalization of trace data.
+type Sanitize func(traces ptrace.Traces) ptrace.Traces
```
Comment (yurishkuro) [inline cmd/jaeger/internal/exporters/storageexporter/sanitizer/sanitizer.go:11]:
```suggestion
type Func func(traces ptrace.Traces) ptrace.Traces
```

File: cmd/jaeger/internal/exporters/storageexporter/sanitizer/emptyservicename.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	return sanitizeEmptyServiceName
+}
+
+func sanitizeEmptyServiceName(traces ptrace.Traces) ptrace.Traces {
+	resourceSpans := traces.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		resourceSpan := resourceSpans.At(i)
+		attributes := resourceSpan.Resource().Attributes()
+		serviceName, ok := attributes.Get(string(otelsemconv.ServiceNameKey))
+		fmt.Printf("service name: %s, ok: %v\n", serviceName.Str(), ok)
```
Comment (yurishkuro) [inline cmd/jaeger/internal/exporters/storageexporter/sanitizer/emptyservicename.go:31]: remove

File: cmd/jaeger/internal/exporters/storageexporter/sanitizer/emptyservicename.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		if !ok {
+			attributes.PutStr(string(otelsemconv.ServiceNameKey), missingServiceName)
+		} else if serviceName.Str() == "" {
```
Comment (yurishkuro) [inline cmd/jaeger/internal/exporters/storageexporter/sanitizer/emptyservicename.go:34]: what if it's not a string attribute?

Comment (mahadzaryab1) [reply cmd/jaeger/internal/exporters/storageexporter/sanitizer/emptyservicename.go]: should we change this check to be the following?
```go
if !ok {
	attributes.PutStr(string(otelsemconv.ServiceNameKey), missingServiceName)
} else if serviceName.Type() != pcommon.ValueTypeStr || serviceName.Str() == "" {
	attributes.PutStr(string(otelsemconv.ServiceNameKey), emptyServiceName)
}
```

Comment (yurishkuro) [reply cmd/jaeger/internal/exporters/storageexporter/sanitizer/emptyservicename.go]: If it's a different type it's a different error message

File: cmd/jaeger/internal/exporters/storageexporter/exporter.go [hunk 5]
```diff
@@ -47,5 +52,5 @@ func (*storageExporter) close(_ context.Context) error {
 }
 
 func (exp *storageExporter) pushTraces(ctx context.Context, td ptrace.Traces) error {
-	return exp.traceWriter.WriteTraces(ctx, td)
+	return exp.traceWriter.WriteTraces(ctx, exp.sanitizer(td))
```
Comment (yurishkuro) [inline cmd/jaeger/internal/exporters/storageexporter/exporter.go:55]: in exporter_test we write a trace with empty resource attributes. Then trace is retrieved from memory storage and verified. You can add a check to that verification confirming that service name attribute was added by the sanitizer.

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**